src/io/ProcessTimer.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * ProcessTimer.h
+ *
+ *  Created on: Apr 12, 2020
+ *      Author: andal
+ */
+
+#ifndef MARDYN_PROCESSTIMER_H
+#define MARDYN_PROCESSTIMER_H
+
+#include <fstream>
+#include <vector>
+#include <array>
+#include <map>
+
+#include "utils/Timer.h"
+
+class ProcessTimer {
+
+public:
+
+    ProcessTimer(){}
+
+    ~ProcessTimer(){}
+
+    //! @brief Inserts new time for given Process-Rank
+    //!
+    //! @param process Rank of process
+    //! @param time Time of given process
+    void insertTime(int process, double time){
+        if( _processes.count(process) > 0)
+            _processes[process].push_back(time);
+        else{
+            std::vector<double> tempVec = {time};
+            _processes.insert(std::pair<int, std::vector<double>>(process, tempVec));
+        }
```

Comment:
`operator []` creates the key-value pair if a key isn't existing yet, so replacing your if-else with only the if-branch should be fine.
```suggestion
        _processes[process].push_back(time);
```